cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaRollerTest.java

```diff
@@ -341,9 +343,37 @@ public void testControllerNeverRollable(VertxTestContext testContext) throws Int
                 emptyList());
     }
 
+    @Test
+    public void testRollHandlesErrorWhenGettingConfig(VertxTestContext testContext) {
+        PodOperator podOps = mockPodOps(podId -> succeededFuture());
+        StatefulSet sts = buildStatefulSet();
+        TestingKafkaRoller kafkaRoller = new TestingKafkaRoller(sts, null, null, podOps,
+                null, null,
+                null, null, new KafkaRoller.ForceableProblem("could not get config exception"),
+            brokerId -> succeededFuture(true), 2);
+        // The algorithm should carry on rolling the pods
+        doSuccessfulRollingRestart(testContext, kafkaRoller,
+                asList(0, 1, 2, 3, 4),
+                asList(0, 1, 3, 4, 2));
+    }
+
+    @Test
+    public void testRollHandlesErrorWhenAlteringConfig(VertxTestContext testContext) {
+        PodOperator podOps = mockPodOps(podId -> succeededFuture());
+        StatefulSet sts = buildStatefulSet();
+        TestingKafkaRoller kafkaRoller = new TestingKafkaRoller(sts, null, null, podOps,
+                null, null,
+                null, new KafkaRoller.ForceableProblem("could not get alter exception"), null,
+            brokerId -> succeededFuture(true), 2);
+        // The algorithm should carry on rolling the pods
+        doSuccessfulRollingRestart(testContext, kafkaRoller,
+                asList(0, 1, 2, 3, 4),
+                asList(0, 1, 3, 4, 2));
+    }
+
```

Comment:
Would be good to see a test for the good case, where the dynamic update can be attempted and is successful, and assert that the broker is not rolled.